incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -530,16 +678,99 @@ private void doMqttReset(
                 doReset(mqtt, originId, routedId, initialId, initialSeq, initialAck, initialMax, traceId);
             }
         }
+
+        public int replyPendingAck()
+        {
+            return (int)(replySeq - replyAck);
+        }
+
+        private int replyWindow()
+        {
+            return replyMax - replyPendingAck();
+        }
+
+        private MqttDataExFW createSubscribeDataEx(
```

Comment:
Try to avoid this pattern of "creating" flyweights before use.

Reply:
What do you mean by "before use"? On both places, we're using it right after it wast "created". Do you mean we should not use a method for this?

Reply:
There is a way to pass a function to the builder so it receives a nested builder without us needing to model is separately and pass by value.